### PR TITLE
CTED-1551 | Added mention permissions on app

### PR DESCRIPTION
**Pull request by JJSwanevelder**

### CODE REVIEW

File: mobilelib.php [hunk 1]
```diff
@@ -156,6 +156,32 @@ function build_allowed_tag_users($userarray) {
...
+ * @param context_course $context
+ * @param $user
+ * @return bool
+ */
+function check_capability($context, $user)
+{
+    global $result, $USER;
+
+    try {
+        if (has_capability('local/getsmarter:mention_' . $user->shortname, $context)) {
```
Comment (BrendonPretorius) [inline mobilelib.php:169]: Could you not just return has_capability('local/getsmarter:mention_' . $user->shortname, $context)? Then you don't need the return false at the end?

**Review by PPrinsloo — APPROVED** [review]

File: mobilelib.php [hunk 2]
```diff
@@ -156,6 +156,32 @@ function build_allowed_tag_users($userarray) {
...
+            return true;
+        }
+    } catch (Exception $e) {
+        error_log($e);
+        $result->result = false;
+        $result->content = $e;
+
+        header('Content-type: application/json');
+        echo json_encode($result);
+        die();
```
Comment (BrendonPretorius) [inline mobilelib.php:179]: Would termination cause any issues down the line? Is it an AJAX call?

File: mobilelib.php [hunk 3]
```diff
@@ -334,11 +360,14 @@ function get_allowed_tag_users($forum_id=0, $group_id=0, $advancedforum=0, $repl
         }
 
         $users = array_merge($users, $course_staff);
+        $context = \context_course::instance($course_id);
 
         if (isset($users)) {
             $data = array();
             foreach ($users as $user) {
-                array_push($data, $user->firstname . ' ' . $user->lastname, $user->userid);
+                if(check_capability($context, $user)) {
```
Comment (BrendonPretorius) [inline mobilelib.php:368]: Follows from question above. What would happen here if an exception is thrown?

File: mobilelib.php (hunk 2 above)
Comment (JJSwanevelder) [reply mobilelib.php:179]: Good call - this was code that was used from web. Believe we can remove the die()

Comment (PPrinsloo) [reply mobilelib.php:179]: Good point, think this was copied from something I wrote and might be leftover debugging code.

File: mobilelib.php (hunk 1 above)
Comment (JJSwanevelder) [reply mobilelib.php:169]: @PPrinsloo think we can do this here and on the web?

Comment (PPrinsloo) [reply mobilelib.php:169]: Sure, I guess. I normally don't return function results directly in Moodle though, they're not super consistent all over the system.

**Review by BrendonPretorius — APPROVED** [review]

**Pull request merged**